core/src/main/scala/org/apache/spark/sql/CachedDataFrame.scala

```diff
@@ -714,13 +715,16 @@ object CachedDataFrame
       // totalParts in runJob.
       var numPartsToTry = 1L
       if (partsScanned > 0) {
-        // If we didn't find any rows after the first iteration, just try all
-        // partitions next. Otherwise, interpolate the number of partitions
-        // we need to try, but overestimate it by 50%.
+        // If we didn't find any rows after the previous iteration, quadruple and retry.
+        // Otherwise, interpolate the number of partitions we need to try, but overestimate
+        // it by 50%. We also cap the estimation in the end.
+        val limitScaleUpFactor = Math.max(session.sessionState.conf.limitScaleUpFactor, 2)
```

Comment:
Can we guard this number to remain positive or zero

Reply:
@vivekwiz The math.max() will make it 2 if negative or zero. Do you mean throw an exception here if the value specified in the conf is <= 1?

Reply:
No I wanted to handle negative value from conf. But yes Max will take care.